core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/BCFile.java

```diff
@@ -672,11 +674,23 @@ public BlockRegion getBlockRegion() {
       }
 
       public void finish() throws IOException {
-        try {
-          in.close();
-        } finally {
-          compressAlgo.returnDecompressor(decompressor);
-          decompressor = null;
+        if (!closed) {
```

Comment:
Whats the purpose of this double check?  Is it an optimization to avoid sync?  I suspect that in the majority of cases where this is called, closed will be false.   Therefore this double check will usually result in an extra round trip to memory (reading closed twice from main memory) in the majority of cases. So this may slow down the majority case.